api/app/src/routes/medical/schemas/shared.ts

```diff
@@ -1,7 +1,27 @@
+import dayjs from "dayjs";
 import { z, ZodString } from "zod";
 
 export const emptyStringToUndefined = (v: string | undefined | null) =>
   v == null || v.length < 1 ? undefined : v;
 
 export const optionalString = (zodSchema: ZodString) =>
   zodSchema.or(z.string().optional()).transform(emptyStringToUndefined);
+
+export function parseToNumericString(str: string): string {
```

Comment:
nit: the name `parse` has been used by us for validation and type conversion, while this function actually changes the data without validating; maybe call it `stripNonNumericChars`, `filterNumbers`, or something?